aten/src/ATen/native/cuda/MixedDtypesMM.cu

```diff
@@ -0,0 +1,256 @@
+                                   // SplitKSerial set to false.
+
+  // Check for current CUTLASS limitations w.r.t. weight sizes.
+  TORCH_CHECK(length_k % 64 == 0 && length_n % 64 == 0,
+              "mixed_dtypes_mm_dispatch_dtype: Number of rows/columns of the "
+              "weight matrix must be divisible by ", 64);
+
+  using ElementAccumulator = float;
+
+  using EpilogueTag = fastertransformer::EpilogueOpBias;
+  using EpilogueOp = typename fastertransformer::Epilogue<
+      ElementOutput,
+      ElementsPerAccessC,
+      ElementAccumulator,
+      EpilogueTag>::Op;
+
+  using DefaultGemmKernel = typename cutlass::gemm::kernel::DefaultGemm<
+      ElementInputA,
+      LayoutInputA,
+      ElementsPerAccessA,
+      ElementInputB,
+      LayoutInputB,
+      ElementsPerAccessB,
+      ElementOutput,
+      LayoutOutput,
+      ElementAccumulator,
+      cutlass::arch::OpClassTensorOp,
+      SmArch,
+      ThreadblockShape,
+      WarpShape,
+      InstructionShape,
+      EpilogueOp,
+      ThreadblockSwizzle,
+      Stages,
+      true,
+      Operator>::GemmKernel;
+  using GemmKernel = cutlass::gemm::kernel::GemmFpAIntB<
+      typename DefaultGemmKernel::Mma,
+      typename DefaultGemmKernel::Epilogue,
+      typename DefaultGemmKernel::ThreadblockSwizzle,
+      SmArch,
+      DefaultGemmKernel::kSplitKSerial>;
+
+  using Gemm = cutlass::gemm::device::GemmUniversalBase<GemmKernel>;
+
+  auto output = input.new_empty({length_m, length_n});
+
+  const auto ldb = length_k * GemmKernel::kInterleave;
+
+  typename Gemm::Arguments arguments(
+      {length_m, length_n, length_k},
+      {(ElementInputA*)input.data_ptr(), length_k},
+      {(ElementInputB*)weight.data_ptr(), ldb},
+      {(ElementInputA*)scale.data_ptr(), 0},
+      {(ElementInputA*)bias.data_ptr(), 0},
+      {(ElementOutput*)output.data_ptr(), length_n},
+      SplitKFactor,
+      {ElementAccumulator(1.f), ElementAccumulator(0.f)});
+
+  Gemm gemm_op;
+
+  cutlass::Status status;
+
+  // Verify that GEMM operation with given arguments can be performed
+  // by CUTLASS.
+  status = gemm_op.can_implement(arguments);
+  CUTLASS_STATUS_CHECK(status);
+
+  // Allocate workspace for CUTLASS mixed datatypes GEMM kernel.
+  const auto workspace_size = Gemm::get_workspace_size(arguments);
+  auto workspace = input.new_empty({(int64_t)workspace_size},
+                                  at::TensorOptions().dtype(at::kByte));
+
+  // Initialize CUTLASS mixed datatypes GEMM object.
+  status = gemm_op.initialize(arguments, workspace.data_ptr(),
+                              at::cuda::getCurrentCUDAStream());
+  CUTLASS_STATUS_CHECK(status);
+
+  // Perform mixed datatypes GEMM operation.
+  status = gemm_op.run(at::cuda::getCurrentCUDAStream());
+  CUTLASS_STATUS_CHECK(status);
+
+  C10_CUDA_KERNEL_LAUNCH_CHECK();
+
+  return output;
+}
+#endif
+
+Tensor
+_fp16_uint8_mm(const Tensor& input, const Tensor& weight, const Tensor& scale,
+               const Tensor& bias) {
```

Comment:
Can we make bias optional? That means generating a specialization with bias fusion enabled and a specialization without bias fusion.